addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -3155,20 +3155,36 @@ registry.SnippetSave = SnippetOptionWidget.extend({
                     classes: 'btn-primary',
                     close: true,
                     click: async () => {
-                        const snippetName = dialog.el.querySelector('.o_we_snippet_name_input').value;
-                        const targetCopyEl = this.$target[0].cloneNode(true);
-                        delete targetCopyEl.dataset.name;
-                        await this._rpc({
-                            model: 'ir.ui.view',
-                            method: 'save_snippet',
-                            kwargs: {
-                                'name': snippetName,
-                                'arch': targetCopyEl.outerHTML,
-                                'template_key': this.options.snippets,
-                                'snippet_class': [...this.$target[0].classList].filter(x => /\bs_./g.test(x))[0],
+                        const save = await this._confirmReload();
+                        if (!save) {
+                            return;
+                        }
+                        // Hack: by the time onSuccess is called after request_save, the
+                        // current widget has been destroyed and is orphaned, so this._rpc
+                        // will not work as it can't trigger_up. To solve this we get a
+                        // reference to the rpc service provider before requesting the save
+                        let serviceProvider = this.getParent();
+                        while (!serviceProvider._call_service) {
+                            serviceProvider = serviceProvider.getParent();
+                        }
```

Comment:
Maybe we can make it a *bit less* hacky by making the `request_save` handler's `onSuccess` call give the rpc function of the service provider as first argument?
So that:
```javascript
this.trigger_up('request_save', {
    onSuccess: async rpc => {
        // ...
        rpc({...});
    }
```

Reply:
What do you think of simply requiring the web.rpc module and using the query method directly? The docstring says:

```js
/**
 * Perform a RPC.  Please note that this is not the preferred way to do a
 * rpc if you are in the context of a widget.  In that case, you should use
 * the this._rpc method.
 */
```

But in this case I'm no longer in the context of a widget so I suppose it's more acceptable and less hacky?